core/src/main/kotlin/com/bumble/appyx/core/navigation/Operation.kt

```diff
@@ -8,6 +8,57 @@ interface Operation<NavTarget, State> :
 
     fun isApplicable(elements: NavElements<NavTarget, State>): Boolean
 
+    /**
+     * Transitions all elements to a new state
+     */
+    fun NavElements<NavTarget, State>.transitionTo(
+        newTargetState: State
+    ) = transitionToIndexed(newTargetState) { _, _ -> true }
+
+    /**
+     * Transitions all elements to a new state based on the individual elements
+     */
+    fun NavElements<NavTarget, State>.transitionTo(
+        newTargetState: (NavElement<NavTarget, State>) -> State
+    ) = map { element ->
+        element.transitionTo(
+            newTargetState = newTargetState.invoke(element),
+            operation = this@Operation
+        )
+    }
+
+    /**
+     * Transitions elements to a new state if they pass a condition based on the element
+     */
+    fun NavElements<NavTarget, State>.transitionTo(
+        newTargetState: State, condition: (element: NavElement<NavTarget, State>) -> Boolean
+    ) = map { element ->
+        if (condition.invoke(element)) {
+            element.transitionTo(
+                newTargetState = newTargetState,
+                operation = this@Operation
+            )
+        } else {
+            element
+        }
+    }
+
+    /**
+     * Transitions elements to a new state if they pass a condition based on the element and its position
+     */
+    fun NavElements<NavTarget, State>.transitionToIndexed(
+        newTargetState: State, condition: (index: Int, element: NavElement<NavTarget, State>) -> Boolean
```

Comment:
Does the condition function need named arguments? In the PR it looks like it could cause confusion with potential variable name shadowing